sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/v1/feedRangeGoneHandler/FeedRangeGoneSplitHandler.java

```diff
@@ -0,0 +1,118 @@
+        // For split, the parent lease will be replaced by the child leases. so we need to remove the current lease in the end.
+        this.removeCurrentLease = true;
+    }
+
+    @Override
+    public Flux<Lease> handlePartitionGone() {
+        //TODO: make sure we are putting feedRangeEpk in the lease token
+        FeedRangeEpkImpl splitRange = (FeedRangeEpkImpl) lease.getFeedRange();
+        AtomicReference<String> min = new AtomicReference<>(splitRange.getRange().getMin());
+        AtomicReference<String> max = new AtomicReference<>(splitRange.getRange().getMax());
+
+        List<String> leaseTokens = new ArrayList<>();
+
+        return Flux.just(overlappingRanges)
+                .flatMapIterable(pkRanges -> {
+
+                    // Create new leases starting from the current min and ending in the current max and across the ordered list of partitions
+                    // Example:
+                    // Current lease epk range: AA-DD
+                    // It split into 3 partition ranges: []-BB, BB-CC, CC-[]
+                    // So we will create EPKRange lease for AA-BB, BB-CC, CC-DD
+
+                    List<FeedRangeEpkImpl> epkRanges = new ArrayList<>();
+                    for (int i = 0; i < pkRanges.size() - 1; i++) {
+                        Range<String> range = pkRanges.get(i).toRange();
+                        epkRanges.add(new FeedRangeEpkImpl(new Range<>(min.get(), range.getMax(), true, false)));
+                        min.set(pkRanges.get(i).getMaxExclusive());
+                    }
+
+                    // add the last range with the original max and the last min
+                    epkRanges.add(new FeedRangeEpkImpl(new Range<>(min.get(), max.get(), true, false)));
+                    return epkRanges;
+                })
+                .flatMap(newEpkRange -> {
+                    // TODO: Annie: what if not all child leases are created
+                    // Should we change to use batch transaction?
+                    String effectiveChildLeaseContinuationToken = this.getEffectiveChildLeaseContinuationToken(
+                            newEpkRange,
+                            this.lease.getContinuationToken());
+
+                    return this.leaseManager
+                            .createLeaseIfNotExist(newEpkRange, effectiveChildLeaseContinuationToken)
+                            .doOnSuccess(newLease -> leaseTokens.add(newLease.getLeaseToken()));
+                })
+                .doOnComplete(() -> {
+                    logger.info(
+                            "Lease with token {} split into {}",
+                            this.lease.getLeaseToken(),
+                            StringUtils.join(leaseTokens, ","));
+                });
+    }
+
+
+    private String getEffectiveChildLeaseContinuationToken(FeedRangeEpkImpl childLeaseFeedRange, String parentLeaseCT) {
```

Comment:
@FabianMeiswinkel/ @kushagraThapar can you help double check the logic here? when split happens, I think we should reconstruct the continuationToken with the token(the lsn) from parent lease, is this the right way?

Reply:
Yes - that is the correct way for splits.

Reply:
If you have any follow-up questions, there Matias has all the context - he implemented the merge support in .Net CFP and merge support in general for .Net - so can definitely help review anything if you need help there.